### PR TITLE
Update changelog for 1.13.0 release

### PR DESCRIPTION
**Pull request by merobi-hub**
### Problem

The changelog is missing changes.

### Solution

This updates the changelog.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

**Review by pawel-big-lebowski — APPROVED** [review]

**Review by pawel-big-lebowski — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +53,10 @@
 * **Spec: Allow nested struct fields in `SchemaDatasetFacet`** [`#2548`](https://github.com/OpenLineage/OpenLineage/pull/2548) [@dolfinus](https://github.com/dolfinus)
     *Allows nested fields support to `SchemaDatasetFacet`.*
 
+### Removed
+* **client/java: [Client] Remove runId from Kafka messageKey** [`#2615`](https://github.com/OpenLineage/OpenLineage/pull/2615) [@dolfinus](https://github.com/dolfinus)  
+    **
```
Comment (pawel-big-lebowski) [inline CHANGELOG.md:58]: is this deliberate?

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +72,8 @@
     *Fixes the run format returned by the `lineage_parent_id` Airflow macro and simplifies the format of the `lineage_parent_id` and `lineage_run_id` macros.*
 * **Dagster: limit Dagster version to 1.6.9** [`#2579`](https://github.com/OpenLineage/OpenLineage/pull/2579) [@JDarDagran](https://github.com/JDarDagran)  
     *Adds an upper limit on supported versions of Dagster as the integration is no longer actively maintained and recent releases introduce breaking changes.*
+* **client/java: [JAVA] fix javadoc for java client** [`#2624`](https://github.com/OpenLineage/OpenLineage/pull/2624) [@pawel-big-lebowski](https://github.com/pawel-big-lebowski)  
+    **
```
Comment (pawel-big-lebowski) [inline CHANGELOG.md:76]: same is here

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +53,10 @@
 * **Spec: Allow nested struct fields in `SchemaDatasetFacet`** [`#2548`](https://github.com/OpenLineage/OpenLineage/pull/2548) [@dolfinus](https://github.com/dolfinus)
     *Allows nested fields support to `SchemaDatasetFacet`.*
 
+### Removed
+* **client/java: [Client] Remove runId from Kafka messageKey** [`#2615`](https://github.com/OpenLineage/OpenLineage/pull/2615) [@dolfinus](https://github.com/dolfinus)
```
Comment (dolfinus) [inline CHANGELOG.md:57]: This item can be removed

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,20 +1,49 @@
...
+* **Spark/Flink: support YAML config files together with `SparkConf` & `FlinkConf` approaches** [`#2583`](https://github.com/OpenLineage/OpenLineage/pull/2583) [@pawel-big-lebowski](https://github.com/pawel-big-lebowski)  
+    *Support config entries being provided by both YAML file and integration specific configuration (`SparkConf`/`FlinkConf`). Allows each integration to have its specific config entries.*
+* **Spark/Flink: job ownership facet** [`#2533`](https://github.com/OpenLineage/OpenLineage/pull/2533) [@pawel-big-lebowski](https://github.com/pawel-big-lebowski)    
+    *Enables configuration entries specifying ownership of the job that will result in an `OwnershipJobFacet` being attached to job facets.*
+* **Java: add `GZIP` compression to `HttpTransport`** [`#2603`](https://github.com/OpenLineage/OpenLineage/pull/2603) [@dolfinus](https://github.com/dolfinus)  
+    *Adds a `compression` option to `HttpTransport` config in the Java client, with `gzip` implementation.*
+* **client/python: [Python] Add GZIP compression to HttpTransport** [`#2604`](https://github.com/OpenLineage/OpenLineage/pull/2604) [@dolfinus](https://github.com/dolfinus)
```
Comment (dolfinus) [inline CHANGELOG.md:29]:
```suggestion
* **Python: Add GZIP compression to HttpTransport** [`#2604`](https://github.com/OpenLineage/OpenLineage/pull/2604) [@dolfinus](https://github.com/dolfinus)  
```

File: CHANGELOG.md (hunk 1 above)
Comment (merobi-hub) [reply CHANGELOG.md:58]: No, thanks for catching. Same as below.

File: CHANGELOG.md (hunk 3 above)
Comment (dolfinus) [reply CHANGELOG.md]: I mean removed. Option `localServerId` -> `messageKey` was removed in #2571, not in #2615.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,44 +1,71 @@
...
 * **Airflow: add `lineage_job_namespace` and `lineage_job_name` macros** [`#2582`](https://github.com/OpenLineage/OpenLineage/pull/2582) [@dolfinus](https://github.com/dolfinus)
     *Adds new Airflow macros `lineage_job_namespace()`, `lineage_job_name(task)` that return an Airflow namespace and Airflow job name, respectively.*
-* **Spec: Allow nested struct fields in `SchemaDatasetFacet`** [`#2548`](https://github.com/OpenLineage/OpenLineage/pull/2548) [@dolfinus](https://github.com/dolfinus)
+* **Spec: allow nested struct fields in `SchemaDatasetFacet`** [`#2548`](https://github.com/OpenLineage/OpenLineage/pull/2548) [@dolfinus](https://github.com/dolfinus)
     *Allows nested fields support to `SchemaDatasetFacet`.*
 
+### Changed
+* **Java: remove `runId` from Kafka messageKey** [`#2615`](https://github.com/OpenLineage/OpenLineage/pull/2615) [@dolfinus](https://github.com/dolfinus)  
+    *With this change, `localServerId` is now **deprecated** and replaced by `messageKey` for all Kafka messages produced via transport.*
+
```
Comment (dolfinus) [inline CHANGELOG.md:53]:
```suggestion
```

File: CHANGELOG.md (hunk 3 above)
Comment (merobi-hub) [reply CHANGELOG.md]: Thanks for catching this, @dolfinus

**Pull request merged**